### ISSUE TITLE
Class jdk.internal.HotSpotIntrinsicCandidate not found - continuing with a stub.

### ISSUE THREAD

**Issue by aryairani** [scala/bug#10959]
I see this error compiling with 2.12.5 or 2.12.6 with `-opt:inline -opt-inline-from:**` with JDK 10.  
```
Class jdk.internal.HotSpotIntrinsicCandidate not found - continuing with a stub.
```
It's okay with JDK 9.  

2.12.4 also fails, but more messily:
```
[error] ## Exception when compiling 35 sources to /Users/arya/unison/runtime-jvm/main/target/scala-2.12/classes
[error] null
[error] scala.tools.asm.ClassReader.<init>(ClassReader.java:185)
[error] scala.tools.asm.ClassReader.<init>(ClassReader.java:168)
[error] scala.tools.nsc.backend.jvm.opt.ByteCodeRepository.$anonfun$parseClass$1(ByteCodeRepository.scala:286)
[error] scala.tools.nsc.backend.jvm.opt.ByteCodeRepository.parseClass(ByteCodeRepository.scala:284)
[error] scala.tools.nsc.backend.jvm.opt.ByteCodeRepository.parsedClassNode(ByteCodeRepository.scala:95)
[error] scala.tools.nsc.backend.jvm.opt.ByteCodeRepository.classNode(ByteCodeRepository.scala:120)
[error] scala.tools.nsc.backend.jvm.BTypesFromSymbols.buildInlineInfo(BTypesFromSymbols.scala:521)
[error] scala.tools.nsc.backend.jvm.BTypesFromSymbols.setClassInfo(BTypesFromSymbols.scala:433)
[error] scala.tools.nsc.backend.jvm.BTypesFromSymbols.classBTypeFromSymbol(BTypesFromSymbols.scala:109)
[error] scala.tools.nsc.backend.jvm.CoreBTypesFromSymbols.$anonfun$_jliMethodHandleRef$1(CoreBTypes.scala:187)
[error] scala.tools.nsc.backend.jvm.BTypes$LazyVar.$anonfun$get$1(BTypes.scala:971)
[error] scala.tools.nsc.backend.jvm.BTypes$LazyVar.get(BTypes.scala:970)
[error] scala.tools.nsc.backend.jvm.CoreBTypesFromSymbols.jliMethodHandleRef(CoreBTypes.scala:186)
[error] scala.tools.nsc.backend.jvm.opt.ByteCodeRepository.isSignaturePolymorphic$1(ByteCodeRepository.scala:193)
[error] scala.tools.nsc.backend.jvm.opt.ByteCodeRepository.findInSuperClasses$1(ByteCodeRepository.scala:202)
[error] scala.tools.nsc.backend.jvm.opt.ByteCodeRepository.$anonfun$methodNode$12(ByteCodeRepository.scala:269)
[error] scala.tools.nsc.backend.jvm.opt.ByteCodeRepository.methodNode(ByteCodeRepository.scala:268)
[error] scala.tools.nsc.backend.jvm.opt.CallGraph.$anonfun$addMethod$2(CallGraph.scala:151)
[error] scala.tools.nsc.backend.jvm.opt.CallGraph.$anonfun$addMethod$2$adapted(CallGraph.scala:148)
[error] scala.collection.Iterator.foreach(Iterator.scala:929)
[error] scala.collection.Iterator.foreach$(Iterator.scala:929)
[error] scala.collection.AbstractIterator.foreach(Iterator.scala:1417)
[error] scala.tools.nsc.backend.jvm.opt.CallGraph.addMethod(CallGraph.scala:148)
[error] scala.tools.nsc.backend.jvm.opt.CallGraph.$anonfun$addClass$1(CallGraph.scala:107)
[error] scala.tools.nsc.backend.jvm.opt.CallGraph.$anonfun$addClass$1$adapted(CallGraph.scala:107)
[error] scala.collection.Iterator.foreach(Iterator.scala:929)
[error] scala.collection.Iterator.foreach$(Iterator.scala:929)
[error] scala.collection.AbstractIterator.foreach(Iterator.scala:1417)
[error] scala.collection.IterableLike.foreach(IterableLike.scala:71)
[error] scala.collection.IterableLike.foreach$(IterableLike.scala:70)
[error] scala.collection.AbstractIterable.foreach(Iterable.scala:54)
[error] scala.tools.nsc.backend.jvm.opt.CallGraph.addClass(CallGraph.scala:107)
[error] scala.tools.nsc.backend.jvm.PostProcessor.$anonfun$runGlobalOptimizations$3(PostProcessor.scala:86)
[error] scala.tools.nsc.backend.jvm.PostProcessor.$anonfun$runGlobalOptimizations$3$adapted(PostProcessor.scala:84)
```

If the problem isn't obvious, let me know and I will try to come up with a minimized example.  Or at least an example.

**Comment by SethTisue** [comment]
looks like a variant of #10717

**Comment by SethTisue** [comment]
@aryairani is this fixed in Scala 2.12.7-bin-f2129b9, which is our current 2.12.7 release candidate? it's available via `resolvers += "scala-integration" at "https://scala-ci.typesafe.com/artifactory/scala-integration/"`

**Comment by SethTisue** [comment]
@aryairani I'll assume this got fixed in 2.12.7 unless you or someone lets us know differently

**Comment by aryairani** [comment]
Sorry for the delay, we have been staying on JDK 9 since encountering this and I haven't had an opportunity to get set up with JDK to try again.
